### ISSUE TITLE
make after make clean fails

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10096]
Migrated from [rt.perl.org#72218](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=72218) (status was 'resolved')

Searchable as RT72218$

**Comment by p5pRT** [comment]
### From @xdg

#### Created by @xdg



make after make clean fails in the Makefile\.PL for the CPAN module\.

Steps to repeat&#8203;:

&nbsp;  $ configure \<your config args here>
&nbsp;  $ make
&nbsp;  $ make test
&nbsp;  $ make clean
&nbsp;  $ make

This appears to be due to overload\.pm trying to load Scalar&#8203;::Util\.

ExtUtils&#8203;::MakeMaker calls Carp&#8203;::longmess\(""\)\, which eventually fails
here&#8203;:

&nbsp;  $arg = defined\($overload&#8203;::VERSION\) ? overload&#8203;::StrVal\($arg\) : "$arg";

Commenting out this line and replacing it with

&nbsp;  $arg = "$arg";

makes the problem go away\.  This is not a good solution\, but I note
it to highlight where the problem occurs\.

Jesse says on IRC that this is a 5\.12 blocker

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.11.3:

Configured by dagolden at Tue Jan 19 15:06:28 EST 2010.

Summary of my perl5 (revision 5 version 11 subversion 3) configuration:
  Commit id: bd12309b58376140a53f4d708bb325f48df1ff70
  Platform:
    osname=linux, osvers=2.6.31-18-generic, archname=x86_64-linux
    uname='linux vulcan 2.6.31-18-generic #55-ubuntu smp fri jan 8 14:54:52 utc 2010 x86_64 gnulinux '
    config_args='-Dusedevel -Dprefix=/opt/perl/fromgit -Dcf_by=dagolden -Dcf_email='dagolden@cpan.org' -Dperladmin='dagolden@cpan.org' -des -Dcc='ccache gcc''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.4.1', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ccache gcc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /lib64 /usr/lib64
    libs=-lnsl -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.10.1.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.10.1'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.11.3:
    /opt/perl/fromgit/lib/site_perl/5.11.3/x86_64-linux
    /opt/perl/fromgit/lib/site_perl/5.11.3
    /opt/perl/fromgit/lib/5.11.3/x86_64-linux
    /opt/perl/fromgit/lib/5.11.3
    .


Environment for perl 5.11.3:
    HOME=/home/david
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=C
    LC_CTYPE=en_US.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=.:~/bin:~/git/utility-scripts:/opt/perl/5.10.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
    PERLPATH=/opt/perl/5.10.1/bin
    PERL_BADLANG (unset)
    PERL_EXTUTILS_AUTOINSTALL=--default-deps
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
@nwc10 - Status changed from 'new' to 'resolved'

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Jan 19\, 2010 at 06&#8203;:26&#8203;:42PM \-0800\, David Golden wrote&#8203;:

> make after make clean fails in the Makefile\.PL for the CPAN module\.
> 
> Steps to repeat&#8203;:
> 
>   $ configure \<your config args here>
>   $ make
>   $ make test
>   $ make clean
>   $ make

> makes the problem go away\.  This is not a good solution\, but I note
> it to highlight where the problem occurs\.
> 

Should be resolved by 603fcd17d4a074139fc76d9d159b3d2d71869bb2

&nbsp;    Remove CPAN's Makefile\.PL; let make\_ext\.pl write one for us\.
&nbsp;    
&nbsp;    Nothing in CPAN's Makefile\.PL was pertinent to building as part of the core
&nbsp;    distribution\, but code within it caused \`make\` after \`make clean\` to fail\.
&nbsp;    Resolves RT \#72218

Nicholas Clark
